packages/next-swc/crates/next-custom-transforms/src/transforms/react_server_components.rs

```diff
@@ -727,6 +728,15 @@ impl ReactServerComponentValidator {
 impl Visit for ReactServerComponentValidator {
     noop_visit_type!();
 
+    // Workaround for swc's parse_program, where it detects empty file as a script.
+    // we may fix upstream to autodetect empty file as a module
```

Comment:
Empty file is a script, and you can try it from vscode by importing an empty file from another file.
vscode will say `foo.js` is not a module